src/OpenTelemetry.AutoInstrumentation/Instrumentations/NLog/TraceContextInjection/Integrations/WriteLogEventToTargetsIntegration.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.NLog.TraceContextInjection.Integrations;
+
+/// <summary>
+/// NLog integration for NLog 6.x.
+/// This integration intercepts NLog's internal WriteLogEventToTargets method to:
+/// 1. Inject trace context (TraceId, SpanId, TraceFlags) into the LogEventInfo properties
+/// 2. Forward log events to OpenTelemetry when the bridge is enabled
+/// </summary>
+/// <remarks>
+/// NLog 6.x renamed the WriteToTargets method to WriteLogEventToTargets.
+/// </remarks>
+[InstrumentMethod(
+    assemblyName: "NLog",
+    typeName: "NLog.Logger",
+    methodName: "WriteLogEventToTargets",
+    returnTypeName: ClrNames.Void,
+    parameterTypeNames: ["NLog.LogEventInfo", "NLog.Internal.ITargetWithFilterChain"],
+    minimumVersion: "6.0.0",
+    maximumVersion: "6.*.*",
+    integrationName: "NLog",
+    type: InstrumentationType.Log)]
+public static class WriteLogEventToTargetsIntegration
```

Comment:
I think that this integration cover only part of the events going through NLog,

There are three `WriteLogEventToTargets` methods:

* `private void WriteToTargets(LogLevel level, Exception? ex, IFormatProvider? formatProvider, string message, object?[]? args)` - covered, it goes through 2 parameter methos
* `private void WriteLogEventToTargets(LogEventInfo logEvent, ITargetWithFilterChain targetsForLevel)` - covered
* `private void WriteLogEventToTargets(Type wrapperType, LogEventInfo logEvent, ITargetWithFilterChain targetsForLevel)` - not covered

The missing part can be reached by public api: `public void Log(Type wrapperType, LogEventInfo logEvent)`